src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,504 @@
+    )
+
+    # Baumgarte derivative term
+    D: jtp.Float = dataclasses.field(
+        default_factory=lambda: jnp.array(0.0, dtype=float)
+    )
+
+    def __hash__(self) -> int:
+        from jaxsim.utils.wrappers import HashedNumpyArray
+
+        return hash(
+            (
+                HashedNumpyArray.hash_of_array(self.mu),
+                HashedNumpyArray.hash_of_array(self.K),
+                HashedNumpyArray.hash_of_array(self.D),
+            )
+        )
+
+    def __eq__(self, other: RigidContactParams) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build(
+        mu: jtp.Float = 0.5,
+        K: jtp.Float = 0.0,
+        D: jtp.Float = 0.0,
+    ) -> RigidContactParams:
+        """Create a `RigidContactParams` instance"""
+        return RigidContactParams(mu=mu, K=K, D=D)
+
+    def valid(self) -> bool:
+        return bool(
+            jnp.all(self.mu >= 0.0)
+            and jnp.all(self.K >= 0.0)
+            and jnp.all(self.D >= 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContactsState(ContactsState):
+    """Class storing the state of the rigid contacts model."""
+
+    def __eq__(self, other: RigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build(**kwargs) -> RigidContactsState:
+        """Create a `RigidContactsState` instance"""
+
+        return RigidContactsState()
+
+    @staticmethod
+    def zero(**kwargs) -> RigidContactsState:
+        """Build a zero `RigidContactsState` instance from a `JaxSimModel`."""
+        return RigidContactsState.build()
+
+    def valid(self, **kwargs) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RigidContactParams = dataclasses.field(
+        default_factory=RigidContactParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    @staticmethod
+    def detect_contacts(
+        W_o_C: jtp.Array,
+        W_o_dot_C: jtp.Array,
+        terrain_height: jtp.Array,
+        terrain_normal: jtp.Array,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+        """
+        Detect contacts between the collidable points and the terrain.
+
+        Args:
+            W_o_C: The position of the collidable points.
+            W_o_dot_C: The linear velocity of the collidable points.
+            terrain_height: The height of the terrain at the collidable point position.
+            terrain_normal: The normal of the terrain at the collidable point position.
+
+        Returns:
+            A tuple containing the activation state of the collidable points along contact normal penetration depth and velocity.
+        """
+
+        # TODO: reduce code duplication with js.contact.in_contact
+        def detect_contact(
+            W_o_C: jtp.Array,
+            W_o_dot_C: jtp.Array,
```

Comment:
Can you use the `W_p_C` variable, so that we keep being consistent with the other contact models? Typically, $\mathbf{o}$ is a symbol used for the origin of a frame, and in this case we don't have an explicit frame, at least from the user point of view.

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e